Add browser support policy for Scribble-generated documentation.

samth opened the pull request:
The short summary: all evergreen browsers from the past year,
plus anything <5 years old with at least 1% share on docs.racket-lang.org.

[Rendered](https://github.com/samth/scribble/blob/browser-support/scribble-lib/scribble/browser-support.md)

samth committed:
Add browser support policy for Scribble-generated documentation.

jackfirth commented on scribble-lib/scribble/browser-support.md:
```diff
@@ -0,0 +1,41 @@
+
+Features that are not needed for reading and understanding the
+documentation, but that are intended for developers of the
+documentation or other specialized audiences, such as links to
```
"such as links to implementation" - is this the motivating use case for this pull request? (Just curious)

sorawee commented:
I think the requirement to have a minimal support for IE6 doesn't make much sense. IE6 by default can't even view sites with https protocol. Even when TLS1.0 is manually enabled, racket-lang.org (and most websites in the world) seems to be serving a content that IE6 doesn't understand, so it will fail with "The page cannot be displayed". This means that users who are still using it can't download Racket in the first place, so there's really no point to support it.

sorawee commented:
FYI: if anyone wants to test IE (including IE6), you can use VMs provided by Microsoft. Unfortunately, Microsoft took these VMs down from their website for unknown reasons, but some users archived them. Use them at your own risk.

- https://gist.github.com/zmwangx/e728c56f428bc703c6f6#gistcomment-3196040
- https://gist.github.com/zmwangx/e728c56f428bc703c6f6#gistcomment-3115797

As noted above, for IE6, we need to enable TLS1.0 to access https websites, but racket-lang.org can't be accessed regardless.

LiberalArtist commented:
FWIW, while I don't care about IE6, I do use `lynx` from time to time, and the docs are quite readable in it.

samth committed:
scribble browser-support: drop IE6 target, cover text-mode browsers

Addresses review feedback on #240: IE 6 can no longer establish a
modern TLS connection to the docs site, so it can't load the pages at
all (sorawee); replace it as the minimal-support target with text-mode
browsers like lynx/links and screen readers (LiberalArtist). Also
generalize the stale June-2020 usage-share example.

LiberalArtist commented on scribble-lib/scribble/browser-support.md:
```diff
@@ -0,0 +1,43 @@
+
+#### Currently supported browsers
+
+- All "evergreen" browsers current within the past 12 months (this
+  includes Chrome, Firefox, Safari, Edge, and other regularly-updated
+  browsers based on similar). "Current" means if that browser was
+  shipped to users within the past 12 months, it is fully supported.
```
Is the intention that this should include supported Firefox ESR releases, since they get point releases (that are not feature updates) more than once per year? https://support.mozilla.org/en-US/kb/firefox-esr-release-cycle

LiberalArtist commented on scribble-lib/scribble/browser-support.md:
```diff
@@ -0,0 +1,43 @@
+  reach the documentation. That is, it is important that the text of
+  the documentation is readable in text-mode browsers such as `lynx`
+  and `links`, in screen readers, and in other browsers with limited
+  styling or scripting. (Very old browsers that can no longer establish
+  a modern TLS connection to the documentation site, such as IE 6, are
+  out of scope: they cannot load the pages at all.)
```
I am not against declaring IE 6 out of scope, but the TLS criterion is about accessing documentation over the network. Isn't IE 6 still able to load locally-installed documentation pages?
